src/lib/db.ts

```diff
@@ -436,7 +436,9 @@ export class DB {
     return new Promise<{document: DBDocument, metadata: DBPublicationDocumentMetadata}>((resolve, reject) => {
       this.createDocument({ type: ProblemPublication, content }).then(({document, metadata}) => {
         const publicationRef = this.firebase.ref(this.firebase.getProblemPublicationsPath(user)).push();
-        const userGroup = groups.groupForUser(user.id)!;
+        // TODO: this is assuming there is always a group for this current user, that doesn't seem
+        // safe???
+        const userGroup = groups.getGroupById(user.currentGroupId)!;
```

Comment:
Did you try it without the `!`? It looks like much of the code below is testing `userGroup` for existence (e.g. lines 442 and 448).

Reply:
Good point. I had to make one change but the types seem happy now.